runelite-client/src/main/java/net/runelite/client/plugins/microbot/playerassist/skill/AttackStyleScript.java

```diff
@@ -0,0 +1,334 @@
+package net.runelite.client.plugins.microbot.playerassist.skill;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.*;
+import net.runelite.api.widgets.ComponentID;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.plugins.microbot.Microbot;
+import net.runelite.client.plugins.microbot.Script;
+import net.runelite.client.plugins.microbot.globval.enums.InterfaceTab;
+import net.runelite.client.plugins.microbot.playerassist.PlayerAssistConfig;
+import net.runelite.client.plugins.microbot.util.combat.Rs2Combat;
+import net.runelite.client.plugins.microbot.util.player.Rs2Player;
+import net.runelite.client.plugins.microbot.util.tabs.Rs2Tab;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static net.runelite.client.plugins.microbot.util.Global.sleepUntil;
+
+
+enum AttackStyle
+{
+    ACCURATE("Accurate", Skill.ATTACK),
+    AGGRESSIVE("Aggressive", Skill.STRENGTH),
+    DEFENSIVE("Defensive", Skill.DEFENCE),
+    CONTROLLED("Controlled", Skill.ATTACK, Skill.STRENGTH, Skill.DEFENCE),
+    RANGING("Ranging", Skill.RANGED),
+    LONGRANGE("Longrange", Skill.RANGED, Skill.DEFENCE),
+    CASTING("Casting", Skill.MAGIC),
+    DEFENSIVE_CASTING("Defensive Casting", Skill.MAGIC, Skill.DEFENCE),
+    OTHER("Other");
+
+    @Getter
+    private final String name;
+    @Getter
+    private final Skill[] skills;
+
+    AttackStyle(String name, Skill... skills)
+    {
+        this.name = name;
+        this.skills = skills;
+    }
+}
+
+@Slf4j
+public class AttackStyleScript extends Script {
```

Comment:
added a disable option if maxed, if maxed and the disable option is not enabled it will just pick any of the attack styles (Please verify, i don't have max combat stats)